support-frontend/assets/pages/[countryGroupId]/checkout.tsx

```diff
@@ -640,55 +674,132 @@ function CheckoutComponent({ geoId }: Props) {
 						>
 							<Box cssOverrides={shorterBoxMargin}>
 								<BoxContents>
-									<PersonalDetails
-										email={email}
-										firstName={firstName}
-										lastName={lastName}
-										isSignedIn={isSignedIn}
-										// TODO - ONE_OFF support, this should be true when ONE_OFF
-										hideNameFields={false}
-										onEmailChange={(email) => {
-											setEmail(email);
-										}}
-										onFirstNameChange={(firstName) => {
-											setFirstName(firstName);
-										}}
-										onLastNameChange={(lastName) => {
-											setLastName(lastName);
-										}}
-										errors={{}}
-										signOutLink={<Signout isSignedIn={isSignedIn} />}
-										contributionState={
-											showStateSelect && (
-												<StateSelect
-													countryId={countryId}
-													state={'STATE'}
-													onStateChange={() => {
-														//  no-op
+									<div css={personalDetailsFieldGroupStyles(true)}>
+										<h2 css={personalDetailsHeader}>Your details</h2>
+										<div>
+											<TextInput
+												id="email"
+												data-qm-masking="blocklist"
+												label="Email address"
+												value={email}
+												type="email"
```

Comment:
can see standard error type `email` used here, presume the `onInvalid` is fired if not matched and the `emailError` displays what is wrong. Nice!

Reply:
@paul-daniel-dempsey [the onInvalid is a standard event](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement/invalid_event) fired via the client.

We tap into this and pipe it through to [Source's implementation of errors](https://github.com/guardian/csnx/blob/main/libs/%40guardian/source-react-components/src/text-input/TextInput.tsx#L65-L68).

As mentioned in the description - it could be a good exercise to use standard DOM libs.